src/Interpreters/TranslateQualifiedNamesVisitor.cpp

```diff
@@ -225,6 +227,69 @@ void TranslateQualifiedNamesMatcher::visit(ASTExpressionList & node, const ASTPt
         }
     }
 
+    auto transform = [&](const ASTPtr & transformer)
+    {
+        if (const auto * apply = transformer->as<ASTColumnsApplyTransformer>())
+        {
+            for (auto & column : node.children)
+            {
+                column = makeASTFunction(apply->func_name, column);
+            }
+        }
+        else if (const auto * except = transformer->as<ASTColumnsExceptTransformer>())
+        {
+            node.children.erase(
+                std::remove_if(
+                    node.children.begin(),
+                    node.children.end(),
+                    [&except](const ASTPtr & node_child)
+                    {
+                        if (const auto * id = node_child->as<ASTIdentifier>())
+                        {
+                            for (const auto & except_child : except->children)
+                            {
+                                if (except_child->as<const ASTIdentifier &>().name == id->shortName())
+                                    return true;
+                            }
+                        }
+                        return false;
+                    }),
+                node.children.end());
+        }
+        else if (const auto * replace = transformer->as<ASTColumnsReplaceTransformer>())
+        {
+            std::map<String, ASTPtr> replace_map;
+            for (const auto & replace_child : replace->children)
+            {
+                if (auto * ast_with_alias = dynamic_cast<ASTWithAlias *>(replace_child.get()))
+                {
+                    if (replace_map.find(ast_with_alias->alias) != replace_map.end())
+                        throw Exception(
+                            "Expressions in columns transformer REPLACE should not contain the same alias more than once",
+                            ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+                    replace_map.emplace(ast_with_alias->alias, replace_child);
+                }
+                else
+                    throw Exception(
+                        "Expressions in columns transformer REPLACE should always contain alias", ErrorCodes::LOGICAL_ERROR);
+            }
+
+            for (auto & column : node.children)
+            {
+                if (const auto * id = column->as<ASTIdentifier>())
+                {
+                    auto replace_it = replace_map.find(id->shortName());
+                    if (replace_it != replace_map.end())
+                    {
+                        auto old_alias = dynamic_cast<const ASTWithAlias *>(id)->alias;
+                        column = replace_it->second;
+                        column->setAlias(old_alias); // TODO What's the correct alias here?
```

Comment:
Would it work?

```
CREATE TEMPORARY TABLE t (x UInt64);
SELECT * REPLACE(x + 1 AS x) REPLACE(x + 1 AS x) FROM t;
```

I expect it to work in this way:

`SELECT (x + 1 AS x) + 1 AS x FROM t`